src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        joinedString += '';
+      } else {
+        joinedString += this[i];
+      };
```

Comment:
do you really need to add blank string? try to invert your if statement and then you wouldn't have to use else
```suggestion
      if (inverted statement) {
        joinedString += this[i];
      }
```

Reply:
yep, ty. that's much much better 👍